### PR TITLE
LED System

### PR DESCRIPTION
**Pull request by firearcher2012**
LED System

### CODE REVIEW

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package org.troyargonauts.subsystems;
+
+import com.ctre.phoenix.led.CANdle;
+import com.ctre.phoenix.led.CANdleConfiguration;
+import com.ctre.phoenix.led.RainbowAnimation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+
+public class LEDSystem extends SubsystemBase {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:9]: Add class description Javadoc

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        candle.configAllSettings(config);
+    }
+
+
+    /**
+     * This method turns the LEDs to a rainbow color pattern. We will generally use this during standby.
+     * @param ledLength refers to the number of LEDs on the strip that will be turned on
+     * @param enable determines if the LEDs are rainbow or not
+     */
+    public void ledStandby(int ledLength, boolean enable) {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:43]: Rename to rainbow() and remove boolean enable parameter

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            candle.setLEDs(goldR, goldG, goldB);
+        }
+    }
+    // Switches color from "Black" (dark gray) to gold
+
+    /**
+     * This method will turn the LEDs purple if we need a purple cube
+     * @param purpleCube determines if the LEDs are purple or not
+     */
+    public void purpleCube(boolean purpleCube) {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:69]: Remove boolean purpleCube parameter

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    // We could maybe set it up so that if we win, these are the lights that will display
+    public void argoColors(boolean win) {
+        if(win) {
+            candle.setLEDs(goldR, goldG, goldB);
+        }
+    }
+    // Switches color from "Black" (dark gray) to gold
+
+    /**
+     * This method will turn the LEDs purple if we need a purple cube
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:66]: Remove "if we need a purple cube"

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+     * @param purpleCube determines if the LEDs are purple or not
+     */
+    public void purpleCube(boolean purpleCube) {
+        if(purpleCube) {
+            candle.setLEDs(purpleR, purpleG, purpleB);
+        }
+    }
+
+    /**
+     * This method will turn the LEDs yellow if we need a yellow cone
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:76]: Remove "if we need a yellow cone"

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+     * @param yellowCone determines if the LEDs are yellow or not
+     */
+    public void yellowCone(boolean yellowCone) {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:79]: Remove boolean yellowCone parameter

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        if(yellowCone) {
+            candle.setLEDs(yellowR, yellowG, yellowB);
+        }
+    }
+
+    /**
+     * This method turns the LEDs off
+     * @param off turns the LEDs off
+     */
+    public void ledOff(boolean off) {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:89]: Remove boolean off parameter

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        if(off) {
+            candle.setLEDs(0, 0, 0);
+        }
+    }
+
+    /**
+     * This method makes the LEDs red. We plan to use this when we lose
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:96]: Remove "We plan to use this when we lose"

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+     * @param lose determines if the LEDs are red or not
+     */
+    public void losingState(boolean lose) {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:99]: Remove boolean lose parameter

**Review by kelmalon — CHANGES_REQUESTED** [review]
Please make the noted changes. Please reach out with any questions. :)

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java (hunk 9 above)
Comment (DarkMG1) [reply src/main/java/org/troyargonauts/subsystems/LEDSystem.java:99]: Do we need this method in the first place? Why do we need a LED state for when we lose?

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+    public final static int purpleB = 106;
+
+    private final CANdle candle;
+    public CANdleConfiguration config;
+
+    /**
+     * This configures the CANdle
+     */
+    public LEDSystem() {
+        candle = new CANdle(5);
```
Comment (DarkMG1) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:31]: Add value to Constants file

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java [hunk 11]
```diff
@@ -54,13 +55,11 @@ public void ledStandby(int ledLength, boolean enable) {
      * This method displays the Argonauts' gold color. We plan to use it if we win.
      * @param win determines if argo colors are on or off
      */
-    // We could maybe set it up so that if we win, these are the lights that will display
     public void argoColors(boolean win) {
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/subsystems/LEDSystem.java:58]: Remove boolean parameter win and if statement

File: src/main/java/org/troyargonauts/Constants.java [hunk 12]
```diff
@@ -9,4 +9,8 @@ public interface Manipulator {
         public static final int ledLength = 300;
     }
 
+    public interface LEDs {
+        int kPort = 5;
```
Comment (kelmalon) [inline src/main/java/org/troyargonauts/Constants.java:13]: Change name to kCANdlePort

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java (hunk 1 above)
Comment (kelmalon) [reply src/main/java/org/troyargonauts/subsystems/LEDSystem.java:12]: Try something like "This class uses REV's CANdle to control the LEDs on the robot"

**Review by kelmalon — CHANGES_REQUESTED** [review]
Looking good! just a couple of things.

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java (hunk 9 above)
Comment (kelmalon) [reply src/main/java/org/troyargonauts/subsystems/LEDSystem.java]: Please remove this method, is not needed.

**Review by kelmalon — APPROVED** [review]

File: src/main/java/org/troyargonauts/subsystems/LEDSystem.java (hunk 9 above)
Comment (kelmalon) [reply src/main/java/org/troyargonauts/subsystems/LEDSystem.java]: @SolidityContract Please remove this method

**Review by kelmalon — COMMENTED** [review]
Please make a separate branch for these and do the changes there.

**Review by kelmalon — APPROVED** [review]
Looking good

**Review by kelmalon — APPROVED** [review]
Looks good thanks!

**Review by DarkMG1 — APPROVED** [review]

**Comment by DarkMG1** [comment]
Already being used on master

**Pull request closed without merging**